cupy/core/_cub_reduction.pyx

```diff
@@ -641,3 +642,22 @@ cdef bint _try_to_call_cub_reduction(
         stream, params, cub_params)
 
     return True
+
+
+# only for testing
+def _get_cub_path():
+    return _cub_path
```

Comment:
can't these be accessed from somewhere else, or just call `_environment` in the tests?

Reply:
No, unfortunately. They are initialized at the module import time, and the subsequent `_environment.get_*_path()` calls won't change them. I tried making them cpdef'd, but realized I hit a Cython limitation that led to this:
https://github.com/cupy/cupy/blob/78a83d71ed2518e7920c2bacd4e9c36b8381b45a/cupy/core/_routines_statistics.pyx#L584-L589

Reply:
Would it be possible to patch the global vars with mock to change the return values?
I don't think that having these setters around is a good idea ...

Reply:
For mock to work, they need to be Python objects, not cdef'd or cpdef'd. In the previous version I simply called `_environment.get_*_path()` every time they are needed so that mock tests work, but I am not a fan either...

Knowing that when nvcc or cub is absent we won't call the cub kernel, how about we just remove the two tests along with the setters?

Reply:
I think its fine to remove those